lib/TPP/ConvertTppToLoops.cpp

```diff
@@ -22,49 +22,98 @@ using namespace mlir::tpp;
 
 namespace {
 
+static bool isScalarVal(Value val) { return !val.getType().isa<ShapedType>(); }
```

Comment:
Shouldn't this be `return isa<ScalarType>()`?

Reply:
I'm pondering on the utility of this function... given this is a really cheap and short check... It makes me thing there's more to it, then I look at the function, and there isn't...

Reply:
Oh it seems to be spir-v specific. Nevermind.

Reply:
It is used in multiple places. I think this is why is an helper function.

Reply:
What I meant is: I don't think the code is large enough to be worth being replaced by a function call.
```
if (!val.getType().isa<ShapedType>())
// vs
if (isScalarVal(val))
```

This is only used in the same pattern:
```
Value scalarLhs =
            isScalarVal(lhs)
                ? lhs
                : b.create<memref::LoadOp>(
                      loc, lhs, getInputOperandIvs(lhs, out, localIvs, zero));
```
and could very easily have been:
```
Value scalarLhs =
            val.getType().isa<ShapedType>()
                ? b.create<memref::LoadOp>(
                      loc, lhs, getInputOperandIvs(lhs, out, localIvs, zero))
                : lhs;
```